js/wp-seo-wordpress-scraper.js

```diff
@@ -0,0 +1,394 @@
+            //document.getElementById("editable-post-name-full").textContent = newUrl;
+            break;
+        default:
+            break;
+    }
+};
+
+/**
+ * Replaces %% strings with WordPress variables
+ * @param textString
+ * @param type
+ * @param object
+ * @param target
+ * @returns {string}
+ */
+YoastSEO_WordPressScraper.prototype.replaceVariables = function( textString, type, object) {
+    if(typeof textString === "undefined"){
+        object[type] = "";
+        this.runDataQueue();
+
+    } else {
+        textString = this.titleReplace(textString);
+        textString = this.defaultReplace(textString);
+        textString = this.parentReplace(textString);
+        textString = this.doubleSepReplace(textString);
+        textString = this.excerptReplace(textString);
+
+        if (textString.indexOf("%%") !== -1 && textString.match(/%%[a-z0-9_-]+%%/i) !== null && typeof this.replacedVars !== "undefined") {
+            var regex = /%%[a-z0-9_-]+%%/gi;
+            var matches = textString.match(regex);
+            for (var i = 0; i < matches.length; i++) {
+                if (typeof( this.replacedVars[matches[i]] ) !== "undefined") {
+                    textString = textString.replace(matches[i], this.replacedVars[matches[i]]);
+                }
+                else {
+                    var replaceableVar = matches[i];
+                    // create the cache already, so we don't do the request twice.
+                    this.replacedVars[replaceableVar] = '';
+                    var srcObj = {};
+                    srcObj.replaceableVar = replaceableVar;
+                    srcObj.textString = textString;
+                    srcObj.type = type;
+                    srcObj.object = object;
+                    this.ajaxReplaceVariables(srcObj);
+                }
+            }
+            if (textString.match(/%%[a-z0-9_-]+%%/i) === null) {
+                object[type] = textString;
+                this.runDataQueue();
+            }
+        } else {
+            object[type] = textString;
+            this.runDataQueue();
+        }
+    }
+};
+
+
+/**
+ * replaces %%title%% with the title
+ * @param textString
+ * @returns textString
+ */
+YoastSEO_WordPressScraper.prototype.titleReplace = function ( textString ){
+    var title = this.analyzerData.title;
+    if(typeof title === "undefined"){
+        title = this.analyzerData.pageTitle;
+    }
+    if (title.length > 0) {
+        textString = textString.replace( /%%title%%/g, title );
+    }
+    return textString;
+};
+
+/**
+ * replaces %%parent_title%% with the selected value from selectbox (if available on page).
+ * @param textString
+ * @returns textString
+ */
+YoastSEO_WordPressScraper.prototype.parentReplace = function ( textString ){
+    var parentId = document.getElementById( "parent_id" );
+
+    if( parentId !== null && parentId.options[parentId.selectedIndex].text !== wpseoMetaboxL10n.no_parent_text ){
```

Comment:
It is not a good idea to check a DOM element for a certain text content, this is very fragile.